ApiDocs.Publishing/CSDL/CsdlWriter.cs

```diff
@@ -0,0 +1,497 @@
+                        AppendToEntityType(edmx, requestTarget, methodCollection);
+                    }
+                    else if (requestTarget.Classification == ODataTargetClassification.NavigationProperty)
+                    {
+                        // TODO: Record somewhere the operations that are available on this NavigationProperty
+                        AppendToNavigationProperty(edmx, requestTarget, methodCollection);
+                    }
+                    else
+                    {
+                        // TODO: Are there interesting things to learn here?
+                        Console.WriteLine("Found type {0}: {1}", requestTarget.Classification, path);
+                    }
+                }
+                catch (Exception ex)
+                {
+                    // TODO: Log this out better than this.
+                    Console.WriteLine("Couldn't serialize request for path {0} into EDMX: {1}", path, ex.Message);
+                    continue;
+                }
+            }
+        }
+
+        private void AppendToNavigationProperty(EntityFramework edmx, ODataTargetInfo navigationProperty, MethodCollection methods)
+        {
+            EntityType parentType = edmx.ResourceWithIdentifier<EntityType>(navigationProperty.QualifiedType);
+
+            NavigationProperty matchingProperty =
+                (from np in parentType.NavigationProperties where np.Name == navigationProperty.Name select np)
+                    .FirstOrDefault();
+            if (null != matchingProperty)
+            {
+                // TODO: Append information from methods into this navigation property
+                StringBuilder sb = new StringBuilder();
+                const string seperator = ", ";
+                sb.AppendWithCondition(methods.GetAllowed, "GET", seperator);
+                sb.AppendWithCondition(methods.PostAllowed, "POST", seperator);
+                sb.AppendWithCondition(methods.PutAllowed, "PUT", seperator);
+                sb.AppendWithCondition(methods.DeleteAllowed, "DELETE", seperator);
+
+                Console.WriteLine("Collection '{0}' supports: ({1})", navigationProperty.QualifiedType + "/" + navigationProperty.Name, sb.ToString());
+            }
+            else
+            {
+                Console.WriteLine(
+                    "EntityType '{0}' doesn't have a matching navigationProperty '{1}' but a request exists for this. Sounds like a documentation error.",
+                    navigationProperty.QualifiedType,
+                    navigationProperty.Name);
+            }
+        }
+
+        /// <summary>
+        /// Use the properties of methodCollection to augment what we know about this entity type
+        /// </summary>
+        /// <param name="requestTarget"></param>
+        /// <param name="methodCollection"></param>
+        private void AppendToEntityType(EntityFramework edmx, ODataTargetInfo requestTarget, MethodCollection methodCollection)
```

Comment:
AppendToNavigationProperty and AppendToEntityType look like they share alot of functionality, namely appending the supported operations with a consistent separator, merge!